p2p/peer.go

```diff
@@ -253,46 +359,69 @@ func (p *peer) SocketAddr() *na.NetAddr {
 }
 
 // Status returns the peer's ConnectionStatus.
-func (p *peer) Status() tcpconn.ConnectionStatus {
-	return p.mconn.Status()
+func (p *peer) Status() any {
+	return p.ConnectionState()
 }
 
-// Send msg bytes to the channel identified by chID byte. Returns false if the
-// send queue is full after timeout, specified by MConnection.
+// Send sends the given envelope via the stream identified by e.ChannelID.
+// Returns false if the send queue is full after 10s.
 //
 // thread safe.
 func (p *peer) Send(e Envelope) bool {
-	return p.send(e.ChannelID, e.Message, p.mconn.Send)
+	streamID := e.ChannelID
+	stream, ok := p.streams[streamID]
+	if !ok {
+		panic(fmt.Sprintf("stream %d not found", streamID))
+	}
+	_ = stream.SetWriteDeadline(time.Now().Add(10 * time.Second))
+	return p.send(e.ChannelID, e.Message, stream.Write)
 }
 
-// TrySend msg bytes to the channel identified by chID byte. Immediately returns
-// false if the send queue is full.
+// TrySend tries to send the given envelope the stream identified by e.ChannelID.
+// Returns false if the send queue is full after 100ms.
 //
 // thread safe.
 func (p *peer) TrySend(e Envelope) bool {
-	return p.send(e.ChannelID, e.Message, p.mconn.TrySend)
+	streamID := e.ChannelID
+	stream, ok := p.streams[streamID]
+	if !ok {
+		panic(fmt.Sprintf("stream %d not found", streamID))
+	}
+	_ = stream.SetWriteDeadline(time.Now().Add(100 * time.Millisecond))
+	return p.send(e.ChannelID, e.Message, stream.Write)
 }
 
-func (p *peer) send(chID byte, msg proto.Message, sendFunc func(byte, []byte) bool) bool {
+func (p *peer) send(streamID byte, msg proto.Message, sendFunc func([]byte) (int, error)) bool {
```

Comment:
You could pass the timeout, for computing the deadline, here as a parameter, so to avoid duplicated code in `Send()` and `TrySend()`.